substrate/frame/contracts/src/tests/mock_network/xcm_utils.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use codec::Encode;
+use core::marker::PhantomData;
+use sp_core::blake2_256;
+use xcm::prelude::*;
+use xcm_executor::traits::ConvertLocation;
+
+/// Prefix for generating alias account for accounts coming
+/// from chains that use 32 byte long representations.
+pub const FOREIGN_CHAIN_PREFIX_PARA_32: [u8; 37] = *b"ForeignChainAliasAccountPrefix_Para32";
+
+/// Prefix for generating alias account for accounts coming
+/// from the relay chain using 32 byte long representations.
+pub const FOREIGN_CHAIN_PREFIX_RELAY: [u8; 36] = *b"ForeignChainAliasAccountPrefix_Relay";
+
+pub struct ForeignChainAliasAccount<AccountId>(PhantomData<AccountId>);
```

Comment:
Why do you have this here? It's already in xcm-builder.
Also, this is a legacy converter, for location conversion, we should be using `HashedDescription` as much as possible, for example: `HashedDescription<AccountId, DescribeFamily<DescribeAllTerminal>>`